TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/subsystems/Lift.java

```diff
@@ -0,0 +1,166 @@
+import static org.firstinspires.ftc.teamcode.robot.constants.LiftConstants.*;
+
+import org.firstinspires.ftc.teamcode.core.lib.gamepad.GamepadManager;
+import org.firstinspires.ftc.teamcode.core.lib.interfaces.Subsystem;
+import org.firstinspires.ftc.teamcode.core.lib.gamepad.SmartGamepad;
+import org.firstinspires.ftc.teamcode.core.lib.pid.PIDController;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Example subsystem that implements the FGCLib.
+ * Look at the example to build your own subsystems
+ */
+public class Lift implements Subsystem {
+    private static Lift instance;
+    private Telemetry telemetry;
+    private DcMotor liftMotorLeft;
+    private DcMotor liftMotorRight;
+    private TouchSensor limitUp;
+    private TouchSensor limitDown;
+    private
+    Map<String ,Double> liftPidPositions = new HashMap<>();
+
+    private SmartGamepad operator;
+    private org.firstinspires.ftc.teamcode.core.lib.pid.PIDController PIDController;
+
+    private Lift() {
+    }
+
+    /**
+     * Initialize method from the Subsystem Interface
+     * @param hardwareMap
+     * @param telemetry
+     */
+    @Override
+    public void initialize(HardwareMap hardwareMap, Telemetry telemetry) {
+        liftMotorRight = hardwareMap.get(DcMotor.class, LIFT_MOTOR_RIGHT);
+        liftMotorLeft = hardwareMap.get(DcMotor.class, LIFT_MOTOR_LEFT);
+        liftMotorLeft.setDirection(DcMotor.Direction.REVERSE);
+        liftMotorRight.setDirection(DcMotor.Direction.FORWARD);
+        limitUp = hardwareMap.get(TouchSensor.class, LIFT_LIMIT_UP);
+        limitDown = hardwareMap.get(TouchSensor.class, LIFT_LIMIT_DOWN);
+        this.telemetry = telemetry;
+
+        liftMotorLeft.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+        liftMotorRight.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+        liftMotorLeft.setMode(DcMotor.RunMode.RUN_WITHOUT_ENCODER);
+        liftMotorRight.setMode(DcMotor.RunMode.RUN_WITHOUT_ENCODER);
+
+        PIDController = new PIDController(PID.kP, PID.kI, PID.kD, PID.kF, POSITION);
+        PIDController.setTolerance(100);
+
+        liftPidPositions.put("Baixo",5712.0);
+        liftPidPositions.put("Medio",7140.0);
+        liftPidPositions.put("Alto", 8568.0);
+
+        telemetry.addData("LIFTSlide Subsystem", "Initialized");
+    }
+
+    /**
+     * Execute method from the Subsystem Interface
+     * @param gamepadManager
+     */
+    @Override
+    public void execute(GamepadManager gamepadManager) {
+        operator = gamepadManager.getOperator();
+
+        telemetry.addData("LIFTSlide Subsystem", "Running");
+
+        controlLIFTSlide();
+
+        if(isLimitDown()){liftPidPositions(0);}
+        
+        if(isLimitUP()){liftPidPositions(0);}
+
+        //Mantem posição após o operador soltar o botão
+        liftPidPositions(PIDController.calculate(liftMotorLeft.getCurrentPosition(),liftMotorLeft.getCurrentPosition()));
```

Comment:
if you need to add a comment, then maybe it's a good idea to add this to a void method with a good name description